speechbrain/utils/run_opts.py

```diff
@@ -0,0 +1,356 @@
+            "--ckpt_interval_minutes",
+            type=float,
+            help="Amount of time between saving intra-epoch checkpoints "
+            "in minutes. If non-positive, intra-epoch checkpoints are not saved.",
+        )
+        parser.add_argument(
+            "--ckpt_interval_steps",
+            type=int,
+            help="Save an intra-epoch checkpoint after this many steps."
+            "If non-positive, intra-epoch checkpoints are not saved.",
+        )
+        parser.add_argument(
+            "--grad_accumulation_factor",
+            type=int,
+            help="Number of batches to accumulate gradients before optimizer step",
+        )
+        parser.add_argument(
+            "--optimizer_step_limit",
+            type=int,
+            help="Number of optimizer steps to run. If not passed, all epochs are run.",
+        )
+        parser.add_argument(
+            "--tqdm_colored_bar",
+            default=False,
+            action="store_true",
+            help="Enable colored progress-bar in tqdm. If this is "
+            "false, tqdm shall use default colors.",
+        )
+        parser.add_argument(
+            "--remove_vector_weight_decay",
+            default=False,
+            action="store_true",
+            help="Make vectors (e.g. norms and biases) a separate parameter group without weight_decay.",
+        )
+        parser.add_argument(
+            "--profile_training",
+            default=False,
+            action="store_true",
+            help=(
+                "If set to True, a profiler will be initiated and tensorboard logs will be generated. "
+                "Please ensure you have installed the torch.TensorBoard profiler with 'pip install torch_tb_profiler'."
+            ),
+        )
+        parser.add_argument(
+            "--profile_warmup",
+            default=5,
+            type=int,
+            help="Number of warmup steps before logging for the profiler.",
+        )
+        parser.add_argument(
+            "--profile_steps",
+            default=5,
+            type=int,
+            help="Number of steps of logging for the profiler",
+        )
+
+        # Accept extra args to override yaml
+        parsed_args, overrides = parser.parse_known_args(arg_list)
+        args_dict = vars(parsed_args)
+        param_file = args_dict.pop("param_file")
+        run_opts = cls(**args_dict)
+
+        overrides = cls._convert_to_yaml(overrides)
+
+        # Checking that DataParallel use the right number of GPU
+        if run_opts.data_parallel_backend and torch.cuda.device_count() == 0:
+            raise ValueError("You must have at least 1 GPU.")
+
+        # force device arg to be the same as local_rank from torchrun
+        local_rank = os.environ.get("LOCAL_RANK")
+        if local_rank is not None and "cuda" in run_opts.device:
+            run_opts.devide = run_opts.devide[:-1] + str(local_rank)
+
+        return param_file, run_opts.as_dict(), overrides

```

Comment:
This should return a `RunOptions` instance for the second object. While it would be cleaner to return *only* a run options object, I think we'll have to keep the return signature for compatibility, so we don't have to update every single recipe.